### PR TITLE
Update maven publish plugin

### PR DESCRIPTION
**Pull request by ZacSweers**
Now supports parallel builds and automatic releasing

https://github.com/vanniktech/gradle-maven-publish-plugin/blob/master/CHANGELOG.md#version-0220-2022-09-09

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -43,7 +43,5 @@ POM_DEVELOPER_URL=https://github.com/slackhq
 POM_INCEPTION_YEAR=2022
 POM_PACKAGING=jar
 SONATYPE_STAGING_PROFILE=com.slack
```
Comment (gabrielittner) [inline gradle.properties:45]: Both POM_PACKAGING and SONATYPE_STAGING_PROFILE are not used anymore

File: gradle.properties [hunk 2]
```diff
@@ -43,6 +43,5 @@ POM_DEVELOPER_URL=https://github.com/slackhq
...
-SONATYPE_HOST=DEFAULT
 RELEASE_SIGNING_ENABLED=true
```
Comment (gabrielittner) [inline gradle.properties:46]: You could also replace this by calling `enableReleaseSigning()` in the DSL

File: gradle.properties (hunk 1 above)
Comment (ZacSweers) [reply gradle.properties:45]: hmm, does it just view whatever is visible to the account being used?

File: gradle.properties (hunk 2 above)
Comment (ZacSweers) [reply gradle.properties:46]: assuming this meant `signAllPublications()`?

Comment (ZacSweers) [reply gradle.properties]: a904f6e

File: gradle.properties (hunk 1 above)
Comment (ZacSweers) [reply gradle.properties]: e96c2e4

Comment (gabrielittner) [reply gradle.properties]: It retrieves the staging profiles in your account and finds the one that equals or is a prefix of the group id

File: gradle.properties (hunk 2 above)
Comment (gabrielittner) [reply gradle.properties]: Yes  👍

File: RELEASING.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ Releasing
 2. Update the `CHANGELOG.md` for the impending release.
 3. `git commit -am "Prepare for release X.Y.Z."` (where X.Y.Z is the new version)
 4. `git tag -a X.Y.Z -m "Version X.Y.Z"` (where X.Y.Z is the new version)
-5. `./gradlew clean publish --no-daemon --no-parallel -x dokkaHtml && ./gradlew closeAndReleaseRepository`
+5. `./gradlew publish -x dokkaHtml`
```
Comment (autonomousapps) [inline RELEASING.md:8]: 😌

**Review by autonomousapps — APPROVED** [review]

**Pull request merged**